server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManagerTests.java

```diff
@@ -231,9 +290,16 @@ public void testDeleteClusterMetadata() throws IOException {
                     + ".dat"
             )
         );
+
+        verify(remoteRoutingTableService).deleteStaleIndexRoutingPaths(List.of(index3Metadata.getUploadedFilename()));
         Set<String> staleManifest = new HashSet<>();
         inactiveBlobs.forEach(blob -> staleManifest.add(new BlobPath().add(MANIFEST_PATH_TOKEN).buildAsString() + blob.name()));
         verify(container).deleteBlobsIgnoringIfNotExists(new ArrayList<>(staleManifest));
+
+        // Test case when remoteRoutingTableService is not enabled
+        remoteRoutingTableService = mock(NoopRemoteRoutingTableService.class);
+        remoteClusterStateCleanupManager.deleteClusterMetadata(clusterName, clusterUUID, activeBlobs, inactiveBlobs);
+        verify(remoteRoutingTableService, never()).deleteStaleIndexRoutingPaths(any());
```

Comment:
Not sure if this is right assertion. `deleteStaleIndexRoutingPaths` should still be called on `NoopRemoteRoutingTableService`

Reply:
Verify can be called only mocked class, hence used the mocked instance.

Reply:
This test is currently passing because remoteClusterStateCleanupManager is never updated with the new mock for remoteRoutingTableService. Once you update that this test will fail, which is expected since the Noop class method should be called.